src/Tribe/Main.php

```diff
@@ -276,6 +276,11 @@ protected function __construct() {
 
 			// let's initialize tec silly-early to avoid fatals with upgrades from 3.x to 4.x
 			add_action( 'plugins_loaded', array( $this, 'plugins_loaded' ), 0 );
+
+			add_action( 'tribe_common_loaded', array( $this, 'bootstrap' ), 0 );
+
+			add_action( 'tribe_common_loaded', array( $this, 'register_addon' ), 5 );
```

Comment:
TEC shouldn't have to register itself as an addon during priority 5 since it isn't an addon, but a base plugin. Ideally, TEC should register itself during the `bootstrap()` process (which it looks like you are doing anyway.